### PR TITLE
Correct API responses and enhance transaction integration tests

### PR DESCRIPTION
**Pull request by Pawel-Mackiewicz**
### Description

This pull request includes the following changes:
- Added comprehensive integration tests to cover transaction history filtering, sorting, pagination, export, and edge cases.
- Refactored integration tests:
  - Introduced utilities for test user and account creation.
  - Centralized common test logic in reusable services.
  - Improved `Transaction` and `ConcurrentTransaction` integration tests for better clarity and maintainability.
  - Corrected API response codes and descriptions in `TransactionHistoryRestControllerInterface`.

### Checklist

- [ ] Code compiles without errors.
- [ ] All tests have been run locally and are passing.
- [ ] Relevant documentation has been updated (if applicable).
- [ ] New tests written and passing (if applicable).
- [ ] Merge conflicts have been addressed and resolved.

## Summary by CodeRabbit

- **Documentation**
  - Improved API documentation for transaction export endpoints to clarify error response codes and descriptions.

- **Tests**
  - Major overhaul and expansion of integration tests for transaction history, including comprehensive coverage for filtering, sorting, pagination, export formats, and authorization scenarios.
  - Introduced utility services for streamlined creation of test users and accounts, ensuring more realistic and maintainable integration tests.
  - Added helper classes for generating unique test user data and valid PESEL numbers for testing purposes.

### CODE REVIEW

File: src/test/java/info/mackiewicz/bankapp/integration/utils/IntegrationTestUserService.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+     * @param hashCode Base hash code for generating the number
+     * @param index    Index to make the number unique
+     *
+     * @return A 9-digit phone number string
+     */
+    private String paddedNumber(int hashCode, int index) {
+        String number = String.valueOf(hashCode + index);
+        int paddingLength = PHONE_NUMBER_LENGTH - number.length();
+        return String.valueOf(PHONE_PADDING_CHAR).repeat(paddingLength) + number;
+    }
```
[the remark this answers is not in the document]
Comment (Pawel-Mackiewicz) [reply src/test/java/info/mackiewicz/bankapp/integration/utils/IntegrationTestUserService.java:84]: it wont. it's for testing purposes.

File: src/test/java/info/mackiewicz/bankapp/integration/TransactionHistoryRestControllerIntegrationTest.java [hunk 2]
```diff
@@ -1,169 +1,493 @@
...
                         .with(SecurityMockMvcRequestPostProcessors.user(testUser))
                         .with(SecurityMockMvcRequestPostProcessors.csrf())
                         .contentType(MediaType.APPLICATION_JSON))
-                .andExpect(status().isOk());
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.content").isArray())
+                .andExpect(jsonPath("$.content.length()").value(0)) // Expect empty array
+                .andExpect(jsonPath("$.totalElements").value(0));
+    }
```
[the remark this answers is not in the document]
Comment (Pawel-Mackiewicz) [reply src/test/java/info/mackiewicz/bankapp/integration/TransactionHistoryRestControllerIntegrationTest.java:447]: no need. it's fine to have more users in db.
every test have different users and accounts - that is what matters.

File: src/test/java/info/mackiewicz/bankapp/integration/TransactionHistoryRestControllerIntegrationTest.java [hunk 3]
```diff
@@ -1,169 +1,493 @@
...
 class TransactionHistoryRestControllerIntegrationTest {
 
-    private static final Integer DEFAULT_ACCOUNT_ID = 7;
     private static final Integer OTHER_ACCOUNT_ID = 42;
+    private static final BigDecimal DEFAULT_BALANCE = new BigDecimal("1000.00");
+    private static final BigDecimal DEFAULT_AMOUNT = new BigDecimal("100.00");
+    private static final String DEFAULT_TRANSACTION_TITLE = "TEST";
+
+    private static int index = 1;
+
```
[the remark this answers is not in the document]
Comment (Pawel-Mackiewicz) [reply src/test/java/info/mackiewicz/bankapp/integration/TransactionHistoryRestControllerIntegrationTest.java:44]: done.

File: src/test/java/info/mackiewicz/bankapp/integration/TransactionHistoryRestControllerIntegrationTest.java [hunk 4]
```diff
@@ -1,169 +1,493 @@
...
 
-        verify(transactionHistoryService).exportTransactions(any(TransactionFilterRequest.class), eq("csv"));
+        // When & Then
+        mockMvc.perform(MockMvcRequestBuilders.get("/api/banking/history")
+                        .param("accountId", OTHER_ACCOUNT_ID.toString())
+                        .with(SecurityMockMvcRequestPostProcessors.user(testUser))
+                        .with(SecurityMockMvcRequestPostProcessors.csrf())
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isForbidden());
     }
```
[the remark this answers is not in the document]
Comment (Pawel-Mackiewicz) [reply src/test/java/info/mackiewicz/bankapp/integration/TransactionHistoryRestControllerIntegrationTest.java:195]: good point. done

File: src/test/java/info/mackiewicz/bankapp/integration/ConcurrentTransactionIntegrationTest.java [hunk 5]
```diff
@@ -320,38 +327,9 @@ void testEdgeCases() {
...
-        return "9".repeat(9 - number.length()) + number;
-    }
-
     private Account createTestAccount(User user) {
-        Account account = accountService.createAccount(user.getId());
         BigDecimal initialBalance = BigDecimal.valueOf(random.nextInt(9000) + 1000);
-        accountService.deposit(account, initialBalance);
-        return account;
+        return testAccountService.createTestAccountWithBalance(user.getId(), initialBalance);
     }
```
[the remark this answers is not in the document]
Comment (Pawel-Mackiewicz) [reply src/test/java/info/mackiewicz/bankapp/integration/ConcurrentTransactionIntegrationTest.java:333]: that doesnt have sense. we never use initialBalance again. @coderabbitai

**Comment by Pawel-Mackiewicz** [comment]
@coderabbitai 
write `squash and merge` commit msg for this PR.

File: src/test/java/info/mackiewicz/bankapp/integration/TransactionHistoryRestControllerIntegrationTest.java [hunk 6]
```diff
@@ -1,169 +1,497 @@
...
-                .andExpect(status().isOk());
+                .andDo(print())
+                .andExpect(status().isOk())
+                // Verify content type for PDF
+                .andExpect(result -> {
+                    String contentType = result.getResponse().getContentType();
+                    // Allow for potential variations like "application/pdf;charset=UTF-8"
+                    assert contentType != null && contentType.startsWith(MediaType.APPLICATION_PDF_VALUE);
+                    assert result.getResponse().getContentAsByteArray().length > 0;
+                });
```
[the remark this answers is not in the document]
Comment (Pawel-Mackiewicz) [reply src/test/java/info/mackiewicz/bankapp/integration/TransactionHistoryRestControllerIntegrationTest.java]: done.
